File: backend/app/db/models/category.py
from typing import Optional

from bson import ObjectId
from odmantic import Index, Model  # Added Index import


class Category(Model):
    """Database model for Categories"""

    name: str
    description: Optional[str] = None
    color: Optional[str] = None  # e.g., #RRGGBB
    user: ObjectId
    is_deleted: bool = False

    model_config = {
        "collection": "categories",
        "indexes": lambda: [
            Index(Category.user, Category.name, Category.is_deleted, unique=True),
        ],
    }

File: frontend/src/types/category.ts
export interface Category {
  id: string;
  name: string;
  description?: string;
  color?: string;
  user_id: string;
  is_deleted: boolean;
}

export interface CategoryCreateRequest {
  name: string;
  description?: string;
  color?: string;
}

export interface CategoryUpdateRequest {
  name?: string;
  description?: string;
  color?: string;
}
